### ISSUE TITLE
[Feature]: Resolve config.plist schema conflicts

### ISSUE THREAD

**Issue by Qonfused** [Qonfused/OCE-Build#1]
The current vision is to use a three-way merge using git, which relies on a common ancestor to apply simultaneous changes. With this approach, the ancestor is the sample plist matching the custom config.plist's OC version, which requires the specific OC version used in that plist. An example of this usage is provided below:

```sh
# Modifies the sample 0.8.8 plist in-place
git merge-file <sample 0.8.8 plist> <sample 0.8.7 plist> <custom 0.8.7 plist>
# Output changes to new plist to preserve sample 0.8.8 plist
git merge-file -p <sample 0.8.8 plist> <sample 0.8.7 plist> <custom 0.8.7 plist> > <new plist>
```

However, there unfortunately is no reliable way to find the OC version from the provided custom config.plist; the most reliable way would be comparing OpenCore.efi hashes from downloaded versions, backtracking from the newer version on a commit-by-commit basis. Instead, enforcing a top-level comment noting the matching opencore version can be enforced to sidestep the need for backtracking.

Additionally, changes in the provided custom config.plist from the matching sample plist can be used to format a yaml file describing changes by tree/property, instead of relying on git for merging changes by line. This may be preferred as an intermediate format for incremental builds, as changes can be readily compared and applied with minimal schema conflict.

**Comment by Qonfused** [comment]
Using a diff also presents another issue, as line changes between files processed by git may include adjacent unchanged lines. This can be resolved by parsing and comparing each plist property included in these conflicts, though a more ideal solution may be in the proposed yaml solution.

**Comment by Qonfused** [comment]
The yaml solution has been implemented in https://github.com/Qonfused/OC-Build/commit/d323df016a79c85949fd25dad183f4a0c0c14b88, though currently there is no implementation yet for schema resolution if a config.plist is specified.

**Comment by Qonfused** [comment]
Superseded by #4. YAML and Plist types are parsed into a native Python dictionary, where keys can be diffed and pruned on a property basis.

Currently, the implementation is pending a full implementation of the pipeline, but this will ultimately be resolved in #4.
